### ISSUE TITLE
FASTA header format

### ISSUE THREAD

**Issue by sjackman** [isovic/racon#61]
I'd suggest using a space rather than an underscore, to separate the sequence identifier from the attributes, and I'd suggest structuring the comment like a SAM header, which is becoming somewhat of a standard.
```
>utg000001l_C:0.998200
```
```
>utg000001l XC:f:0.998200
```

**Comment by rvaser** [comment]
I need to somehow modify the sequence name because I parse them to the first space and if you run multiple rounds of error correction of reads, the newly corrected reads will have the same names as the raw ones (I am discarding any duplicate sequences to save memory). Should I add a character to the end of the name plus the space and SAM tag?

**Comment by sjackman** [comment]
I've been using Racon to polish Miniasm contigs. The input sequences are the unitigs and the raw reads, then in the second round the polished unitigs and the raw reads. So for my purpose, I don't need to change the unitig names.

When you're polishing reads, the input is the raw reads and all-to-all alignments in the first round. In the second round, is it the polished reads and the raw reads, again all-to-all alignments? Okay. I think I understand the problem. I like your suggestion of adding a suffix to the sequence name. Perhaps…
```
>utg000001lx XC:f:0.998200
```
where `x` is perhaps `r` for Racon, `c` for corrected, or `p` for polished?
Then a second round of polishing would be
…
```
>utg000001lxx XC:f:0.998200
```
I would make this suffix an option with `--suffix=x`, and I'd suggest making the default no suffix, so `--suffix=''`, or if the default is non-empty make it possible to specify `--no-suffix`.

**Comment by rvaser** [comment]
Is it necessary to make it optional? Or do I even need to add the suffix? How often are there multiple rounds of read error correction? :D

**Comment by sjackman** [comment]
No, I don't think adding a suffix is needed, and a separate tool can be used to add a suffix to each sequence if necessary.

**Comment by rvaser** [comment]
Isn't it necessary to modify the sequence identifier to distinguish the polished from the unpolished ones (identifier being the string up to first whitespace character)? Is there any reason you are suggesting the parameter to be optional?

**Comment by sjackman** [comment]
When I'm polishing miniasm unitigs, I'd prefer that the unitig identifiers not be modified. It's not necessary to distinguish the polished from the unpolished unitigs in this use case.

**Comment by rvaser** [comment]
Should I add the suffix only in error correction? I don't like adding too much parameters :D

**Comment by sjackman** [comment]
That seems like reasonable default behaviour to me. What suffix are you thinking of adding?

**Comment by rvaser** [comment]
Just lowercase r for racon after each iteration (`>utg000001lr` -> `>utg000001lrr` ...).

**Comment by sjackman** [comment]
Works for me

**Comment by sjackman** [comment]
I have a feature request, that would be quite useful to me!

[Bandage](https://rrwick.github.io/Bandage/) will draw the thickness of the unitig segments according to their depth when the segments include the read count (`RC:i`) attribute. See https://github.com/GFA-spec/GFA-spec/blob/master/GFA1.md#optional-fields-2. Could Racon please add the number of reads that were used to polish a unitig to its FASTA header? For example
```
>utg000001lx RC:i:123 XC:f:0.998200
```
I'd then convert this file to a GFA file. Bandage can also read FASTA files. It's possible that it could read this coverage information directly from the FASTA file without needing to create a GFA file.

**Comment by sjackman** [comment]
While we're chatting about the FASTA header, I sometimes find adding the length in characters useful `LN:i:12345`, particularly since Racon likely changed the length of the sequence. Thoughts on including it? The argument against it is that it's redundant. I don't feel strongly about it. Just thought I'd mention it.

**Comment by rvaser** [comment]
I'll add the requested tags to headers soon.

**Comment by rvaser** [comment]
Does the order of tags matter? Do the custom ones (those starting with X|Y|Z) need to be at end? Also, the RC tag will be approximate, i.e. I will count overlaps per sequence and some sequences might have multiple overlaps with the same target or even with itself. Is that fine with you?

**Comment by sjackman** [comment]
An approximate count is fine. I use it with Bandage to estimate the copy number of the segments. Close is good enough.

> Does the order of tags matter? Do the custom ones (those starting with X|Y|Z) need to be at end?

Order shouldn't matter, but often the custom tags appear at the end.

**Comment by rvaser** [comment]
Latest commit produces headers with LN, RC and XC tags. I'll create a release once I test everything.

**Comment by sjackman** [comment]
Thanks, Robert! 🏅

**Comment by rvaser** [comment]
Released in `1.2.0`.

**Comment by sjackman** [comment]
Here's a PR to update Brewsci/bio: https://github.com/brewsci/homebrew-bio/pull/104
